packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -364,22 +364,24 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
       return;
     final double oldPage = page;
     _viewportFraction = value;
+
     if (oldPage != null)
-      forcePixels(getPixelsFromPage(oldPage));
+      // This part should only be reacheable when it will definitely trigger a rebuild.
+      correctPixels(getPixelsFromPage(oldPage));
```

Comment:
this feel dangerous to me. That is true when you change this property in didupdatewidget. If you are certain that is the case for the future as well, you should add an assert to verify that.

Reply:
I am not sure if this is better but how about
```
if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
     correctPixels(getPixelsFromPage(oldPage));
    } else {
      forcePixels(getPixelsFromPage(oldPage));
    }

Reply:
This class itself is private, and the setter is only used in `PageController.attach`. I'm not sure if we should call `notifyListeners` in `ScrollController.attach`, I think it should be up to the caller?

Reply:
I am more worry about we change the scroll offset but not markneedbuild, and yes this should be fine because this is private class and as of now it only called during the build stage. However, it is possible this might change. If you think that is unlikely, we should still add an assert here to guard it.

Reply:
Yeah let me get rid of the setter altogether. The implementation should probably live in `PageController.attach` which has more context.